third_party/2and3/jinja2/utils.pyi

```diff
@@ -1,26 +1,51 @@
-from typing import Any, Callable, Iterable, Optional
+import sys
+from typing import Any, Callable, IO, Iterable, Optional, Text, TypeVar, Union
 
 from markupsafe import Markup as Markup, escape as escape, soft_unicode as soft_unicode
 
 missing: Any
 internal_code: Any
 concat: Any
 
-def contextfunction(f): ...
-def evalcontextfunction(f): ...
-def environmentfunction(f): ...
-def internalcode(f): ...
-def is_undefined(obj): ...
+if sys.version_info >= (3, 6):
+    from builtins import _PathLike
+    _PathType = Union[bytes, Text, _PathLike]
+else:
+    _PathType = Union[bytes, Text]
+if sys.version_info >= (3,):
+    text_type = str
+else:
+    text_type = unicode
+
+_T = TypeVar('_T')
+
+class ContextFunction():
```

Comment:
These don't exist at runtime, so they should be prefixed with `_`.

Also, the current way these are typed is not quite ideal because we lose type information for the argument types. I haven't tried this, but what if you do the following:
```
_CallableT = TypeVar("_CallableT", bound=Callable[..., Any])

class _ContextFunction(Protocol[_CallableT]):
    contextfunction: Literal[True]
    __call__: _CallableT

def contextfunction(f: _CallableT) -> _ContextFunction[_CallableT]: ...
```

Reply:
I'm not sure that matters? We're not type checking the body of the function using this code, just code that uses these APIs.